core-services/src/main/java/org/zalando/nakadi/partitioning/HashPartitionStrategy.java

```diff
@@ -31,32 +31,29 @@ public HashPartitionStrategy(final HashPartitionStrategyCrutch hashPartitioningC
     }
 
     @Override
-    public String calculatePartition(final EventType eventType, final JSONObject event, final List<String> partitions)
+    public String calculatePartition(final EventType eventType,
+                                     final JSONObject jsonEvent,
+                                     final List<String> partitions)
+            throws PartitioningException {
+        final var partitioningData = getDataByPartitionKeyFields(eventType, jsonEvent);
+
+        return calculatePartition(partitioningData, partitions);
+    }
+
+    private String calculatePartition(final PartitioningData partitioningData, final List<String> partitions)
             throws InvalidPartitionKeyFieldsException {
-        final List<String> partitionKeyFields = eventType.getPartitionKeyFields();
-        if (partitionKeyFields.isEmpty()) {
+        if (partitioningData.getPartitionKeys().isEmpty()) {
```

Comment:
Maybe check for `null` here, even if it's a private method.

Reply:
We resolved this conversation offline